src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -0,0 +1,267 @@
+package io.supertokens.inmemorydb.queries;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.supertokens.inmemorydb.Start;
+import io.supertokens.inmemorydb.config.Config;
+import io.supertokens.inmemorydb.ConnectionWithLocks;
+import io.supertokens.pluginInterface.RowMapper;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPUsedCode;
+
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.execute;
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.update;
+
+public class TOTPQueries {
+    public static String getQueryToCreateUsersTable(Start start) {
```

Comment:
manually test what happens if we do a select for update on the used_codes table, and then we issue a delete query (on a different terminal) in the getTotpUsersTable table for that user (for mysql and postgresql).

Reply:
Postgres:
While txn1 does `SELECT FOR UPDATE` and hasn't commited, txn2 isn't allowed to update/delete any of the rows selected by txn. It . Furthermore, txn2 can read the original rows but it cannot see any updates in the same rows done by txn1 unless txn1 commits them. But txn2 can insert new rows without any issue (and txn1 will be able to see them it if does another SELECT query in the same txn)

MySQL:

Reply:
TODO: Check what happens on a conlict? If both txns insert same values. Whichever commits late is rejected?

Reply:
To discuss

Reply:
Discussed. It will wait.